data-prepper-api/src/main/java/org/opensearch/dataprepper/model/codec/OutputCodec.java

```diff
@@ -56,10 +55,11 @@ public interface OutputCodec {
      */
     String getExtension();
 
-    default Event addTagsToEvent(Event event, String tagsTargetKey) throws JsonProcessingException {
-        String eventJsonString = event.jsonBuilder().includeTags(tagsTargetKey).toJsonString();
-        Map<String, Object> eventData = objectMapper.readValue(eventJsonString, new TypeReference<>() {
-        });
-        return JacksonLog.builder().withData(eventData).build();
+    default String buildJsonString(Event event, String tagsTargetKey, List<String> includeKeys, List<String> excludeKeys) throws JsonProcessingException {
```

Comment:
I don't think we should have this method here. Only the JSON codec cares about building a JSON string. So each codec should handle this independently.